src/main/java/de/hysky/skyblocker/skyblock/dungeon/puzzle/boulder/BoulderBoard.java

```diff
@@ -0,0 +1,127 @@
+     */
+    public BoulderObject getObjectAtPosition(int x, int y) {
+        return isValidPosition(x, y) ? grid[x][y] : null;
+    }
+
+    /**
+     * Retrieves the 3D position of the BoulderObject at the specified position on the board.
+     *
+     * @param x The x-coordinate of the position.
+     * @param y The y-coordinate of the position.
+     * @return The BlockPos representing the 3D position of the BoulderObject,
+     * or null if no object is present at the specified position.
+     */
+    public BlockPos getObject3DPosition(int x, int y) {
+        BoulderObject object = getObjectAtPosition(x, y);
+        return (object != null) ? object.get3DPosition().offset(Direction.Axis.Y, -1) : null;
+    }
+
+    /**
+     * Places a BoulderObject at the specified position on the board.
+     *
+     * @param x      The x-coordinate of the position.
+     * @param y      The y-coordinate of the position.
+     * @param object The BoulderObject to place on the board.
+     */
+    public void placeObject(int x, int y, BoulderObject object) {
+        grid[x][y] = object;
+    }
+
+    public int getHeight() {
+        return height;
+    }
+
+    public int getWidth() {
+        return width;
+    }
+
+    /**
+     * Checks whether the specified position is valid within the bounds of the game board.
+     *
+     * @param x The x-coordinate of the position to check.
+     * @param y The y-coordinate of the position to check.
+     * @return {@code true} if the position is valid within the bounds of the board, {@code false} otherwise.
+     */
+    private boolean isValidPosition(int x, int y) {
+        return x >= 0 && y >= 0 && x < height && y < width;
+    }
+
+    /**
+     * Generates a character array representation of the game board.
+     * Each character represents a type of BoulderObject or an empty space.
+     *
+     * @return A 2D character array representing the game board.
+     */
+    public char[][] getBoardCharArray() {
+        char[][] boardCharArray = new char[height][width];
+        for (int x = 0; x < height; x++) {
+            for (int y = 0; y < width; y++) {
+                BoulderObject boulderObject = grid[x][y];
+                boardCharArray[x][y] = (boulderObject != null) ? boulderObject.type().charAt(0) : '.';
+            }
+        }
+        return boardCharArray;
+    }
+
+    /**
+     * Prints the current state of the game board to the console.
+     * Each character represents a type of BoulderObject or an empty space.
+     */
+    public void printBoard() {
+        for (int x = 0; x < height; x++) {
+            for (int y = 0; y < width; y++) {
+                BoulderObject boulderObject = grid[x][y];
+                String displayChar = (boulderObject != null) ? boulderObject.type() : ".";
+                System.out.print(displayChar);
+            }
+            System.out.println();
+        }
+    }
```

Comment:
Doesn't really matter but maybe something like `Waterboard#boardToString` would be better since `System.out.print` is not hardcoded.